src/Http/Routing/test/UnitTests/Internal/LinkGenerationDecisionTreeTest.cs

```diff
@@ -340,6 +340,385 @@ public void SelectMultipleEntries_BothMatch_OrderedByTemplate()
+
+            var entry2 = CreateMatch(new { page = "/Store/Buy", area = "Admin", controller = (string)null, action = (string)null, });
+            entry2.Entry.RouteTemplate = TemplateParser.Parse("b");
+            entries.Add(entry2);
+
+            var tree = new LinkGenerationDecisionTree(entries);
+
+            var context = CreateContext(new { page = "/Store/Buy", }, new { controller = "Home", action = "Index", area = "Admin", });
+
+            // Act
+            var matches = tree.GetMatches(context.Values, context.AmbientValues).Select(m => m.Match).ToList();
+
+            // Assert
+            Assert.Collection(
+                matches,
+                m => { Assert.Same(entry2, m); });
+        }
+
+        [Fact]
+        public void GetMatches_LinkToPageFromController_WithActionValue()
+        {
+            // Arrange
+            var entries = new List<OutboundMatch>();
+
+            var entry1 = CreateMatch(new { controller = "Home", action = "Index", area = (string)null, page = (string)null, });
+            entry1.Entry.RouteTemplate = TemplateParser.Parse("a");
+            entries.Add(entry1);
+
+            var entry2 = CreateMatch(new { page = "/Store/Buy", area = (string)null, controller = (string)null, action = (string)null, });
+            entry2.Entry.RouteTemplate = TemplateParser.Parse("b");
+            entries.Add(entry2);
+
+            var tree = new LinkGenerationDecisionTree(entries);
+
+            var context = CreateContext(new { page = "/Store/Buy", action = "buy", }, new { controller = "Home", action = "Index", page = "16", });
+
+            // Act
+            var matches = tree.GetMatches(context.Values, context.AmbientValues).Select(m => m.Match).ToList();
+            
+            // Assert
+            Assert.Collection(
+                matches,
+                m => { Assert.Same(entry2, m); });
+        }
+
+        [Fact]
+        public void GetMatches_LinkToPageFromController_WithActionValueAmbiguous()
+        {
+            // Arrange
+            var entries = new List<OutboundMatch>();
+
+            var entry1 = CreateMatch(new { controller = "Home", action = "Index", area = (string)null, page = (string)null, });
+            entry1.Entry.RouteTemplate = TemplateParser.Parse("a");
+            entries.Add(entry1);
+
+            var entry2 = CreateMatch(new { page = "/Store/Buy", area = (string)null, controller = (string)null, action = (string)null, });
+            entry2.Entry.RouteTemplate = TemplateParser.Parse("b");
+            entries.Add(entry2);
+
+            var tree = new LinkGenerationDecisionTree(entries);
+
+            var context = CreateContext(new { page = "/Store/Buy", action = "Index", }, new { controller = "Home", action = "Index", page = "16", });
+
+            // Act
+            var matches = tree.GetMatches(context.Values, context.AmbientValues).Select(m => m.Match).ToList();
+
+            // Assert
+            Assert.Empty(matches);
+        }
+
```

Comment:
Note: I didn't make any attempt to address the problem of writing a conventional route that uses `{page}` as a parameter. I'm not sure this can really be solved.